core/src/scene/directionalLight.cpp

```diff
@@ -19,10 +19,19 @@ void DirectionalLight::setDirection(const glm::vec3 &_dir) {
     m_direction = glm::normalize(_dir);
 }
 
-void DirectionalLight::setupProgram( std::shared_ptr<ShaderProgram> _shader ) {
+void DirectionalLight::setupProgram(const std::shared_ptr<View>& _view, std::shared_ptr<ShaderProgram> _shader ) {
+
+    m_direction_eye = m_direction;
+    if (m_origin != LightOrigin::CAMERA) {
+        glm::mat3 normalMatrix = glm::mat3(_view->getViewMatrix()); // Transforms surface normals into camera space
+        normalMatrix = glm::transpose(glm::inverse(normalMatrix));
+
+        m_direction_eye = normalMatrix * m_direction;
```

Comment:
This looks like it gives the right results for world-origin lights, but it's kind of a waste to generate the same normal matrix so many times! The tile object itself calculates the same normal matrix and makes it available as 'u_normalMatrix' in our shaders. I think it makes sense to just use this matrix in the shader if the light is world-space-fixed.